### DISCUSSION TITLE
Multiple Email Accounts / Multiple Sender Addresses

### DISCUSSION THREAD

**Discussion by TheFox** [craftcms/cms#8830] [Ideas]
Hello,
is there a way to have multiple Email accounts? In the Admin Email settings domain.tld/admin/settings/email it looks like I can only use one account for sending emails. What's the best practice for providing multiple email addresses?

For example I have two different emails. For each email the user should be able to answer to 2 different email addresses. Is it better to use two different email accounts, or just one email account and two different Reply-To header in the emails which are sent to the users?

I use Craft CMS 3.3.20.1.

**Comment by brandonkelly** [comment]
It’s currently not possible to specify multiple senders from Settings → Email, however you can do it by [overriding the system mailer component](https://docs.craftcms.com/v3/config/app.html#mailer-component) in `config/app.php`:

```php
return [
    'components' => [
        'mailer' => function() {
            // Instantiate the Mailer component with the system settings
            $settings = craft\helpers\App::mailSettings();
            $mailer = Craft::createObject(craft\helpers\App::mailerConfig($settings));

            // Override the `from` property with multiple senders
            $mailer->from = [
                'a@domain.tld' => 'Name A',
                'b@domain.tld' => 'Name B',
            ];

            // Return it
            return $mailer;
        },
    ],
];
```

(If you’d prefer to have multiple Reply-To emails instead, set `replyTo` instead of `from` on the mailer.)

We can consider adding support for multiple From/Reply-To emails alongside #4594, so I’ll leave this open and put it on the 3.5 list.

**Reply by mofman** [reply]
Is this still working? I've copied and pasted it exactly as it is and desite the email settings panel displaying "It looks like these settings are being overridden by config/app.php." I'm still getting the default from names/emails in the CMS.

**Reply by brianjhanson** [reply]
Are you on Craft 3 or 4? If you're running 4, the config has changed slightly. [Check out the docs for the latest](https://craftcms.com/docs/4.x/config/app.html#mailer). 

Make sure to heed the warning directly below that as well, if you're testing via the `Settings -> Email` page, custom configuration in `app.php` will not be respected. The [`mailer/test`](https://craftcms.com/docs/4.x/console-commands.html#mailer) command will respect settings in the `app.php` though.

**Reply by mofman** [reply]
> Make sure to heed the warning directly below that as well, if you're testing via the `Settings -> Email` page, custom configuration in `app.php` will not be respected. The [`mailer/test`](https://craftcms.com/docs/4.x/console-commands.html#mailer) command will respect settings in the `app.php` though.

In that case, that is what the problem is; might be worth while displaying this to user when they're using that test button and they have overridden these settings. 

Thanks for quick reply!

**Comment by mylesthedev** [comment]
🤞 this will be in 3.5, just realized I need it for an up and coming site launch...

**Comment by jannisborgers** [comment]
How is the status on this one? We’re launching a multi-site installation where all (sister-)sites have a different email address next to the contact form and it would be great to be able to send from these addresses instead of the one defined in email settings.

**Comment by brandonkelly** [comment]
@jannisborgers Didn’t make the cut for 3.5 unfortunately. See my last comment for an example of how you can override the `mailer` component though.
